File: code-samples/ai/interactions.py
import os,sys
import logging
import requests
from ringcentral import SDK
from dotenv import load_dotenv

# Load Enviroment variables
load_dotenv()

# Invoke Interaction Analysis API 
def analyzeInteractions():
    # Endpoint to invoke Interaction analysis API 
    endpoint = os.getenv('RC_SERVER_URL')+"/ai/insights/v1/async/analyze-interaction"

    # Webhook as Query string
    querystring = {"webhook":os.getenv('WEBHOOK_ADDRESS')}

    # Payload
    payload = {
        "contentUri": "https://github.com/suyashjoshi/ringcentral-ai-demo/blob/master/public/audio/sample1.wav?raw=true",
        "encoding": "Wav",
        "languageCode": "en-US",
        "source": "RingCentral",
        "audioType": "Meeting",
        "insights": ["All"],
        "enableVoiceActivityDetection": True,
        "enablePunctuation": True,
        "enableSpeakerDiarization": False
    }

    try:
        # Instantiate Ringcentral SDK 
        rcsdk = SDK( os.getenv('RC_CLIENT_ID'),os.getenv('RC_CLIENT_SECRET'),os.getenv('RC_SERVER_URL'))
        platform = rcsdk.platform()

        # Login Using JWT
        platform.login( jwt=os.getenv('RC_JWT') );

        # Make HTTP POST call to the Interaction analysis endpoint with the query string and payload
        response = platform.post(endpoint, payload, querystring);
        print(response.json());

    except Exception as e:  
        print(e)

try:
    analyzeInteractions()
except Exception as e:
    print(e)

File: code-samples/ai/interactions.js
const RC = require('@ringcentral/sdk').SDK;
require('dotenv').config();

MEDIA_URL   = process.env.RC_MEDIA_URL;
WEBHOOK_URL = '<INSERT YOUR WEBHOOK URL>';

// Initialize the RingCentral SDK and Platform
const rcsdk = new RC({
    'server':       process.env.RC_SERVER_URL,
    'clientId':     process.env.RC_CLIENT_ID,
    'clientSecret': process.env.RC_CLIENT_SECRET
});

const platform = rcsdk.platform();

// Login into the Developer Portal using Developer's JWT Credential
platform.login({
    'jwt': process.env.RC_JWT
});

// Call the Interaction Analysis API right after login asynchronously
platform.on(platform.events.loginSuccess, () => {
    analyzeInteraction();
})

async function analyzeInteraction() {
    try {
        let resp = await platform.post("/ai/insights/v1/async/analyze-interaction?webhook=" + WEBHOOK_URL,{
            "contentUri":                   MEDIA_URL,
            "encoding":                     "Wav",
            "languageCode":                 "en-US",
            "source":                       "RingCentral",
            "audioType":                    "Meeting",
            "insights":                     [ "All" ],
            "enableVoiceActivityDetection": true,
            "enablePunctuation":            true,
            "enableSpeakerDiarization":     false
        });
        console.log("Job is " + resp.statusText + " with HTTP status code " + resp.status);
    } 
    catch (e) {
        console.log("An Error Occurred : " + e.message);
    }
}
